pandas/core/internals/blocks.py

```diff
@@ -484,6 +484,7 @@ def split_and_operate(
         -------
         list of blocks
         """
+        assert isinstance(self.values, np.ndarray)
```

Comment:
need to ensure split_and_operate is not called from EA block (or base method not overridden)

Reply:
move assert after ndim check